Src/FluentAssertions/Execution/MessageBuilder.cs

```diff
@@ -48,23 +48,25 @@ private string SubstituteIdentifier(string message, string identifier, string fa
 
             message = Regex.Replace(message, pattern, match =>
             {
-                string defaultIdentifier = match.Groups["default"].Value;
-                string result = "object";
-
+                const string result = " ";
                 if (!identifier.IsNullOrEmpty())
                 {
-                    result = identifier;
+                    return result + identifier;
```

Comment:
You know i hate those intermediate `return`s, don't you?

Reply:
I was not aware of that.
I usually code in a "return early, return often" style and a prefer Guard Clauses.

My intention here was to avoid the initialization of `defaultIdentifier` if the first branch was taken.

With a single `return` it could either be the code below, which I dislike because of additional indentation, or extracting the else part into a separate method.
```c#
if (!identifier.IsNullOrEmpty())
{
    result = identifier;
}
else
{
    string defaultIdentifier = match.Groups["default"].Value;

    if (!defaultIdentifier.IsNullOrEmpty())
	{
		result = defaultIdentifier;
	}
	else if (!fallbackIdentifier.IsNullOrEmpty())
	{
		result = fallbackIdentifier;
	}
}
```

Reply:
To be fair, I don't like that either. So it's a bit of a taste thing. I've been bitten a few times by long methods that suddenly return somewhere half-way. Of course, the problem wasn't really the `return` statement but in the length of that method.